encoding/blockchain/blockchain.go

```diff
@@ -91,6 +94,78 @@ func WriteVarstr31(w io.Writer, str []byte) (int, error) {
 	return n + n2, err
 }
 
+// WriteVarstrList writes a varint31 length prefix followed by the
+// elements of l as varstrs.
+func WriteVarstrList(w io.Writer, l [][]byte) (int, error) {
+	n, err := WriteVarint31(w, uint64(len(l)))
+	if err != nil {
+		return n, err
+	}
+	for _, s := range l {
+		n2, err := WriteVarstr31(w, s)
+		n += n2
+		if err != nil {
+			return n, err
+		}
+	}
+	return n, err
+}
+
+// ReadVarstrList reads a varint31 length prefix followed by that many
+// varstrs.
+func ReadVarstrList(r io.Reader) ([][]byte, int, error) {
+	nelts, n, err := ReadVarint31(r)
+	if err != nil {
+		return nil, n, err
+	}
+	if nelts == 0 {
+		return nil, n, nil
+	}
+	result := make([][]byte, 0, nelts)
+	for ; nelts > 0; nelts-- {
+		s, n2, err := ReadVarstr31(r)
+		n += n2
+		if err != nil {
+			return nil, n, err
+		}
+		result = append(result, s)
+	}
+	return result, n, nil
+}
+
+// WriteExtensibleString sends the output of the given function,
+// together with a varint31 length prefix, to w.
+func WriteExtensibleString(w io.Writer, f func(io.Writer) error) (int, error) {
```

Comment:
Can/should we document that WriteExtensibleString always provides a *bytes.Buffer to f, and that the signature of f is this way mainly so that it matches io.WriterTo? That way, f can safely ignore errors from its Writer (which is guaranteed to be a *bytes.Buffer, which never returns an error from Write). That would let us clean up some (or maybe just one) of the call sites.